src/config/config.go

```diff
@@ -65,40 +65,46 @@ func ProcessArgs() *Config {
 	flag.StringVar(&ret.HTTPSProxy, "https-proxy", "", "A proxy URL to use for creating HTTPS connections outside the cluster")
 	flag.StringVar(&ret.NoProxy, "no-proxy", "", "A comma-separated list of destination domain names, domains, IP addresses, or other network CIDRs to exclude proxying")
 	flag.StringVar(&ret.ServiceIPs, "service-ips", "", "All IPs of assisted service node")
-	flag.StringVar(&ret.HighAvailabilityMode, "high-availability-mode", "Full", "high-availability expectations. default is 'Full', which represents the behavior in a \"normal\" cluster. Use 'None' for single-node deployment")
+	flag.StringVar(&ret.HighAvailabilityMode, "high-availability-mode", "", "high-availability expectations, \"Full\" which represents the behavior in a \"normal\" cluster. Use 'None' for single-node deployment. Leave this value as \"\" for workers as we do not care about HA mode for workers.")
 	flag.BoolVar(&ret.CheckClusterVersion, "check-cluster-version", false, "Do not monitor CVO")
 	flag.StringVar(&ret.MustGatherImage, "must-gather-image", "", "Custom must-gather image")
 	flag.Var(&ret.DisksToFormat, "format-disk", "Disk to format. Can be specified multiple times")
 
 	var installerArgs string
 	flag.StringVar(&installerArgs, "installer-args", "", "JSON array of additional coreos-installer arguments")
-
 	h := flag.Bool("help", false, "Help message")
 	flag.Parse()
 
-	if ret.NoProxy != "" {
-		utils.SetNoProxyEnv(ret.NoProxy)
+	if err := ret.SetInstallerArgs(installerArgs); err != nil {
+		printHelpAndExit(err)
 	}
+
 	if h != nil && *h {
-		printHelpAndExit()
+		printHelpAndExit(nil)
+	}
+
+	if ret.NoProxy != "" {
+		utils.SetNoProxyEnv(ret.NoProxy)
 	}
 
+	ret.SetDefaults()
+	return ret
+}
+
+func (c *Config) SetInstallerArgs(installerArgs string) error {
 	if installerArgs != "" {
-		err := json.Unmarshal([]byte(installerArgs), &ret.InstallerArgs)
-		if err != nil {
-			println(err.Error())
-			printHelpAndExit()
-		}
+		return json.Unmarshal([]byte(installerArgs), &c.InstallerArgs)
 	}
-	if ret.HighAvailabilityMode == models.ClusterHighAvailabilityModeNone {
-		validRoles := []string{string(models.HostRoleMaster), string(models.HostRoleBootstrap)}
-		if !funk.ContainsString(validRoles, ret.Role) {
-			println("high-availability-mode is set to None, but host role is %s. should be one of: %s", ret.Role, validRoles)
-			printHelpAndExit()
-		}
+	return nil
+}
+
+func (c *Config) SetDefaults() {
+	if c.Role == string(models.HostRoleWorker) {
```

Comment:
Why do we need this if it's already taken care of by the service?
https://github.com/openshift/assisted-service/pull/3718/files
Can we keep the role validation instead?

Reply:
The main issue here is that UserManagedNetworking can be set to "None" when you are adding a working.
This occurs when attempting to add a Day2 worker to an existing SNO cluster.

The original solution was to take care of this earlier in the process by not passing the parameter to the installer, however this has proved to be problematic as it will require a breaking change to the service and there is presently no good mechanism for ensuring that all users are running the correct version of the agent.

After discussion with Avishay and Omer, we decided to deal with the issue here instead.